eventhubsprocessor/eh_partition_pump.py

```diff
@@ -0,0 +1,131 @@
+    Pulls and messages from lease partition from eventhub and sends them to processor
+    """
+    def __init__(self, host, lease):
+        PartitionPump.__init__(self, host, lease)
+        self.eh_client = None
+        self.partition_receiver = None
+        self.partition_receive_handler = None
+
+    async def on_open_async(self):
+        """
+        Eventhub Override for on_open_async
+        """
+        _opened_ok = False
+        _retry_count = 0
+        while (not _opened_ok) and (_retry_count < 5):
+            try:
+                await self.open_clients_async()
+                _opened_ok = True
+            except Exception as err:
+                logging.warning("%s,%s PartitionPumpWarning: Failure creating client or receiver,\
+                                retrying: %s", self.host.guid, self.partition_context.partition_id,
+                                repr(err))
+                last_exception = err
+                _retry_count += 1
+
+        if not _opened_ok:
+            await self.processor.process_error_async(self.partition_context, last_exception)
+            self.set_pump_status("OpenFailed")
+
+        if self.pump_status == "Opening":
+            self.set_pump_status("Running")
+            self.eh_client.run_daemon()
+            await self.partition_receiver.run()
+
+        if self.pump_status == "OpenFailed":
+            self.set_pump_status("Closing")
+            await self.clean_up_clients_async()
+            self.set_pump_status("Closed")
+
+    async def open_clients_async(self):
+        """
+        Responsible for establishing connection to event hub client
+        throws EventHubsException, IOException, InterruptedException, ExecutionException
+        """
+        await self.partition_context.get_initial_offset_async()
+        # Create event hub client and receive handler and set options
+ 
+        self.partition_receive_handler = AsyncReceiver()
+        self.eh_client = EventHubClient(self.host.eh_connection_string) \
+                        .subscribe(self.partition_receive_handler,
+                                   self.partition_context.consumer_group_name,
+                                   self.partition_context.partition_id,
+                                   Offset(self.partition_context.offset))
+        self.partition_receiver = PartitionReceiver(self)
+
+    async def clean_up_clients_async(self):
+        """
+        Resets the pump swallows all exceptions
+        """
+        if self.partition_receiver:
+            if self.eh_client:
+                self.eh_client.stop()
+                self.partition_receiver = None
+                self.partition_receive_handler = None
+                self.eh_client = None
+
+    async def on_closing_async(self, reason):
+        """
+        Overides partition pump on cleasing
+        """
+        await self.clean_up_clients_async()
+
+class PartitionReceiver:
+    """
+    Sends events from a async until lease is lost
```

Comment:
Receives events from an async receiver until lease is lost.

Reply:
Good catch this is leftover from when there was a queue that we were writing too. Will update